repos/system_upgrade/common/actors/reportleftoverpackages/reportleftoverpackages/libraries/reportleftoverpackages.py

```diff
@@ -0,0 +1,47 @@
+from leapp import reporting
+from leapp.libraries.stdlib import api
+from leapp.models import LeftoverPackages, RemovedPackages
+
+
+def process():
+    removed_packages = next(api.consume(RemovedPackages), None)
+    leftover_packages = next(api.consume(LeftoverPackages), LeftoverPackages())
+    to_remove = ['-'.join([pkg.name, pkg.version, pkg.release]) for pkg in leftover_packages.items]
+
+    if removed_packages:
+        title = 'Leftover RHEL packages have been removed'
+
+        if removed_packages.items:
+            removed = ['-'.join([pkg.name, pkg.version, pkg.release]) for pkg in removed_packages.items]
+            reporting.create_report([
+                reporting.Title(title),
+                reporting.Summary('Following packages have been removed:\n{}'.format('\n'.join(removed))),
+                reporting.Severity(reporting.Severity.HIGH),
+                reporting.Groups([reporting.Groups.SANITY]),
+            ] + [reporting.RelatedResource('package', pkg.name) for pkg in removed_packages.items])
+        else:
+            summary = ('Following packages have been removed:\n'
+                       '{}\n'
+                       'Dependent packages may have been removed as well, please check that you are not missing '
+                       'any packages.\n'.format('\n'.join(to_remove)))
```

Comment:
I know this isn't your change, but while we are refactoring, we have somewhat standard way for separating the items in a report, an example is here along with how to use it: https://github.com/oamg/leapp-repository/blob/f3913f7b8ee11f5e67354bf18963f05269bb511c/repos/system_upgrade/common/actors/checkosrelease/libraries/checkosrelease.py#L7

Reply:
Good to know, I applied it.